src/raw_stream.rs

```diff
@@ -142,10 +142,7 @@ impl RawDevice {
     /// Paths are typically something like `/dev/input/event0`.
     #[inline(always)]
     pub fn open(path: impl AsRef<Path>) -> io::Result<RawDevice> {
-        Self::_open(path.as_ref())
-    }
-
-    fn _open(path: &Path) -> io::Result<RawDevice> {
```

Comment:
This is here to minimize monomorphization overhead. I don't mind refactoring this a bit to allow `fd` rework, but please retain the minimal public `AsRef<Path>` interface.

Reply:
I was wondering why you were inlining it anyway. Now I understand. I will refactor this asap.